### ISSUE TITLE
Add Symptomatic State to ABM

### ISSUE THREAD

**Issue by xsaschako** [DLR-SC/memilio#688]
### Feature description

Quoting @DavidKerkmann from #681

> We remove the InfectionStates Exposed and Recovered from the ABM. Removing Recovered was already planned, as there is no difference to Susceptible if we only look at the symptoms. Removing Exposed leads to every newly infected person moving to InfectedNoSymptoms immediately and they do have a chance to infect people as of the current infectivity model.
> Adding a threshold for being able to infect others is an option, but I commend that this should be handled in addition to this change to account for the needs of comparison to other models or the needs of any external party. In this regard, I suggest moving the infectivity function to the parameters, so it can be set individually for any virus (type).
> Going further, an idea would be to rename InfectionState to SymptomaticState or similar, and considering renaming of some of the attributes too, as there is no complete 1:1 relation to the InfectionStates of the ODE anyway. For example, we could have NoSymptoms, InfectedNoSymptoms, InfectedSymptoms, InfectedSevere, InfectedCritical, Dead. Looking at this, we can even combine the first two states, as we would only track symptoms with this.

### Checklist

- [X] Attached labels, especially loc:: or model:: labels.
- [X] Linked to project.

**Comment by jubicker** [comment]
As a proposal: A possibility to link the infectivity to the symptomatic state would be to adjust the time scale of the infectivity curve according to the symptomatic states the person will pass. For example, if a person has a mild infection course, the time that the person is infected will be shorter compared to a person that has a severe infection course and goes to hospital. As it is implemented now, both persons have the same infectivity curve, so the person with a mild infection course could have peak infectivity at one time point and be recovered one hour later. To make that more realistic, we could compress or stretch the time scale of the infectivity curve according to the time a person is infected. Like that the overall shape of the curve would remain, but infectivity would however be dependent on the individual infection course.

**Comment by DavidKerkmann** [comment]
Thank you for your proposal! Keep in mind that there is no shown correlation between symptoms and viral load/infectiousness. It might be that a person with mild symptoms is more infectious than a person that suffers from severe symptoms. Could you clarify some of your phrases? What do you mean by `mild infection course`? Does this refer to symptoms or to viral load? Also, how do you define the `time a person is infected`?

**Comment by jubicker** [comment]
Yes as there is no correlation between viral load and symptoms, the course of the infectivity curve will stay the same independent of the infection course, only the time is scaled. By a mild infection course I refer to the symptoms e.g. persons that will not go to hospital. I would define the time a person is infected as the time the function `is_infected` (used to calculate the exposure rates in the interaction method in location.cpp) is true. This is currently the case when a person's infection state is neither susceptible nor recovered. So for a person that goes to hospital this time is `t_Exposed + t_InfectedNoSymptomsToSymptoms + t_InfectedSymptomsToSevere + t_InfectedSevereToRecovered` whereas for a person with a mild infection this is `t_Exposed + t_InfectedNoSymptomsToRecovered`.
